corporate/lib/stripe.py

```diff
@@ -1498,6 +1613,31 @@ def update_context_initial_upgrade(self, context: Dict[str, Any]) -> None:
             "demo_organization_scheduled_deletion_date"
         ] = self.realm.demo_organization_scheduled_deletion_date
 
+    @override
+    def downgrade_at_the_end_of_billing_cycle(self, plan: Optional[CustomerPlan] = None) -> None:
+        if plan is None:
+            plan = get_current_plan_by_realm(self.realm)  # nocoverage
+        assert plan is not None
+        do_change_plan_status(plan, CustomerPlan.DOWNGRADE_AT_END_OF_CYCLE)
+
+    # During realm deactivation we instantly downgrade the plan to Limited.
+    # Extra users added in the final month are not charged. Also used
+    # for the cancellation of Free Trial.
+    @override
+    def downgrade_now_without_creating_additional_invoices(
+        self,
+        plan: Optional[CustomerPlan] = None,
+    ) -> None:
+        if plan is None:
+            plan = get_current_plan_by_realm(self.realm)
+            if plan is None:
+                return
+
+        self.process_downgrade(plan)
+        plan.invoiced_through = LicenseLedger.objects.filter(plan=plan).order_by("id").last()
+        plan.next_invoice_date = next_invoice_date(plan)
+        plan.save(update_fields=["invoiced_through", "next_invoice_date"])
```

Comment:
I think these two functions be defined on the abstract `BillingSession` instead, if you just change them to find the current plan using `get_current_plan_by_customer` instead. Could be fine to do as a second commit if that feels cleaner to you, but this seems like a code path that shouldn't need to be forked.

Reply:
Yeah, makes sense. Defined the two function in `BillingSession`